### PR TITLE
Properly replace expired token with new token

### PR DESCRIPTION
**Pull request by saraOSI**

### CODE REVIEW

**Comment by saraOSI** [comment]
@brunoOSI @ratneshraval Can you please review?

**Comment by ratneshraval** [comment]
@saraOSI it should check new token validity before replacing the session token.

File: app/controllers/sandbox_base_controller.rb [hunk 1]
```diff
@@ -1,24 +1,23 @@
...
-  token = ''
-
-
   protected
 
   def authenticate_with_cwds
-    if session[:token].blank?
-      if Cwds::Authentication.token_validation(params[:token], ENV['AUTHENTICATION_API_BASE_URL'])
-        session[:token] = params[:token]
+    token = params[:token]
```
Comment (saraOSI) [inline app/controllers/sandbox_base_controller.rb]: Perry uses the token on the URL and that can cause more troubles in the future.
A request for a change on that behavior would be requested to TPT team.

**Comment by saraOSI** [comment]
@ratneshraval The code is checking the validity before setting the new token, as far as I can tell.

**Pull request merged**